### PR TITLE
XYZ parser

### PR DESCRIPTION
**Pull request by Hadegda**
## Description

Add parser for XYZ file format.

## Motivation and Context

Adds ability to reading information from xyz files.

## How Has This Been Tested?

Have been tested with mocha's test.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/functional/data/complexTestData.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+  formats: ['pdb', 'cif', 'mmtf', 'xyz'],
   num: {
     atoms: 327,
     bonds: 337,
-    residues: 46,
+    residues: {xyz: 1, default: 46},
     chains: 1,
-    molecules: 1,
-    units: 2,
+    molecules: {xyz: 0, default: 1},
```
Comment (paulsmirnov) [inline test/functional/data/complexTestData.js]: Please generate one molecule in your parser (include everything there by default).

File: src/io/parsers/XYZParser.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  constructor(data, options) {
+    super(data, options);
+
+    this._complex = null;
+    this._atomsInf = null;
+
+    this._options.fileType = 'xyz';
+  }
+
+  canParse(data, options) {
```
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/io/parsers/XYZParser.js]: Thanks, change is accepted

File: src/io/parsers/XYZParser.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    if (!data) {
+      return false;
+    }
+    return typeof data === 'string' && Parser.checkDataTypeOptions(options, 'xyz');
+  }
+
+  canProbablyParse(data) {
```
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/io/parsers/XYZParser.js]: Thanks, change is accepted

File: src/io/parsers/XYZParser.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  _parseAtomsInf() {
+    const het = true;
+    const altLoc = ' ';
+    const occupancy = 1;
+    const tempFactor = 1;
+    const charge = 0;
+
+    const chain = this._complex.addChain(' ');
+    const residue = chain.addResidue('LIG', 1, ' ');
```
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/io/parsers/XYZParser.js]: Thanks, change is done

File: test/functional/data/complexTestData.js (hunk 1 above)
Comment (Hadegda) [reply test/functional/data/complexTestData.js]: Thanks, change is done

File: src/io/parsers/XYZParser.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  _parseToAtomsInf(source) {
+    const endnAtoms = source.indexOf('\n');
+    const nAtoms = parseInt(source.substring(0, endnAtoms), 10);
+    const endComment = source.indexOf('\n', endnAtoms + 1);
+    const comment = source.slice(endnAtoms + 1, endComment);
+
+    const startAtomsInf = endComment + source.substring(endComment).search(/\S/);
+    this._atomsInf = source.substring(startAtomsInf).split(/[\s,]*\n[\s,]*/);
```
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/io/parsers/XYZParser.js]: Thanks, for your comment 
I checked parsing time for big file (5J37)
Parsing xyz file takes the similar time as parsing pdb or cif files

File: src/io/parsers/XYZParser.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+  canParse(data, options) {
+    if (!data) {
+      return false;
+    }
+    return typeof data === 'string' && Parser.checkDataTypeOptions(options, 'xyz');
+  }
+
+  canProbablyParse(data) {
+    return _.isString(data) && /^\s*data_/i.test(data);
```
[the remark this answers is not in the document]
Comment (Hadegda) [reply src/io/parsers/XYZParser.js]: Thanks, change is accepted

**Pull request merged**